jaxrs/src/main/java/io/airlift/jaxrs/AsyncResponseBinder.java

```diff
@@ -0,0 +1,110 @@
+package io.airlift.jaxrs;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import io.airlift.units.Duration;
+import jakarta.servlet.AsyncEvent;
+import jakarta.servlet.AsyncListener;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.ws.rs.container.AsyncResponse;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.Response;
+
+import java.util.Optional;
+import java.util.concurrent.Executor;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import static io.airlift.jaxrs.AsyncResponseHandler.bindAsyncResponse;
+import static jakarta.ws.rs.core.Response.status;
+import static java.util.Objects.requireNonNull;
+
+public class AsyncResponseBinder
+{
+    private final HttpServletRequest servletRequest;
+    private boolean cancelOnClientDisconnect = true;
+    private Optional<Duration> timeout = Optional.empty();
+    private Function<Duration, Supplier<Response>> timeoutResponse = AsyncResponseBinder::defaultTimeoutResponse;
+
+    private AsyncResponseBinder(HttpServletRequest servletRequest)
+    {
+        this.servletRequest = requireNonNull(servletRequest, "servletRequest is null");
+    }
+
+    public AsyncResponseBinder cancelOnClientDisconnection(boolean cancelOnClientDisconnection)
+    {
+        this.cancelOnClientDisconnect = cancelOnClientDisconnection;
+        return this;
+    }
+
+    public AsyncResponseBinder withTimeout(Duration timeout)
+    {
+        this.timeout = Optional.of(timeout);
+        return this;
+    }
+
+    public AsyncResponseBinder withTimeoutResponse(Response response)
+    {
+        requireNonNull(response, "response is null");
+        this.timeoutResponse = timeout -> () -> response;
+        return this;
+    }
+
+    public AsyncResponseBinder withTimeoutResponse(Supplier<Response> responseSupplier)
+    {
+        requireNonNull(responseSupplier, "responseSupplier is null");
+        this.timeoutResponse = timeout -> responseSupplier;
+        return this;
+    }
+
+    public void bindTo(AsyncResponse response, ListenableFuture<?> future, Executor httpExecutor)
```

Comment:
If we do the above the `AsyncResponse response` parameter here would not be necessary.